src/_P001_Switch.ino

```diff
@@ -142,10 +142,9 @@ boolean Plugin_001(byte function, struct EventStruct *event, String& string)
         //String command = parseString(string, 2);
         //String strPar1 = parseString(string, 3);
 
-        String tempstring = string;
-        tempstring.toLowerCase();
         // returns pin value using syntax: [plugin#gpio#pinstate#xx]
-        if (tempstring.startsWith(F("gpio,pinstate")))
+        string.toLowerCase();
+        if (string.startsWith(F("gpio,pinstate")))
```

Comment:
I don't get it, now it is again changing `string` ?

Reply:
yes, but 'string' is already a copy of the String from misc.ino and string get changed anyway in the PLUGIN_REQUEST.
So I assume is safe to change it.

Reply:
Nope it is not safe to change it there.
You may only change it when the call is successful. If the call is not successful, the next plugin is queried and that one may need the case sensitive info.

But you can do a case insensitive compare.